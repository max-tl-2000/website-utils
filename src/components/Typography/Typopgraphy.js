/*
 * Copyright (c) 2022 Reva Technology Inc., all rights reserved.
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Licensed under the Elastic License 2.0; you may not use this file except
 * in compliance with the Elastic License 2.0.
 */

import { createTextElement } from './creator';

export const Text = createTextElement('Text');
export const Title = createTextElement('Title');
export const Header = createTextElement('Header');
export const Caption = createTextElement('Caption');
export const Link = createTextElement('Link');
