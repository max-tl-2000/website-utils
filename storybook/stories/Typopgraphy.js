/*
 * Copyright (c) 2022 Reva Technology Inc., all rights reserved.
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Licensed under the Elastic License 2.0; you may not use this file except
 * in compliance with the Elastic License 2.0.
 */

/* eslint-disable react/prefer-stateless-function */
import React, { Component } from 'react';
import { storiesOf } from '@storybook/react';
import Field from '../../src/components/Field/Field';
import Block from '../helpers/Block';
import * as T from '../../src/components/Typography/Typopgraphy';

class Wrapper extends Component {
  render() {
    return (
      <Block>
        <Field noMargin>
          <T.Header>Header: The quick brown fox jumps over the lazy dog</T.Header>
        </Field>
        <Field noMargin>
          <T.Text>Text: The quick brown fox jumps over the lazy dog</T.Text>
        </Field>
        <Field noMargin>
          <T.Caption>Title: The quick brown fox jumps over the lazy dog</T.Caption>
        </Field>
        <Field noMargin>
          <T.Header secondary>Header: The quick brown fox jumps over the lazy dog</T.Header>
        </Field>
        <Field noMargin>
          <T.Text secondary>Text: The quick brown fox jumps over the lazy dog</T.Text>
        </Field>
        <Field noMargin>
          <T.Caption secondary>Title: The quick brown fox jumps over the lazy dog</T.Caption>
        </Field>

        <Field noMargin>
          <T.Header highlight>Header: The quick brown fox jumps over the lazy dog</T.Header>
        </Field>
        <Field noMargin>
          <T.Text highlight>Text: The quick brown fox jumps over the lazy dog</T.Text>
        </Field>
        <Field noMargin>
          <T.Caption highlight>Title: The quick brown fox jumps over the lazy dog</T.Caption>
        </Field>
        <Field noMargin>
          <T.Header error>Header: The quick brown fox jumps over the lazy dog</T.Header>
        </Field>
        <Field noMargin>
          <T.Text error>Text: The quick brown fox jumps over the lazy dog</T.Text>
        </Field>
        <Field noMargin>
          <T.Caption error>Title: The quick brown fox jumps over the lazy dog</T.Caption>
        </Field>

        <div style={{ background: 'black', padding: '5px 20px' }}>
          <Field noMargin>
            <T.Header lighterForeground>Header: The quick brown fox jumps over the lazy dog</T.Header>
          </Field>
          <Field noMargin>
            <T.Text lighterForeground>Text: The quick brown fox jumps over the lazy dog</T.Text>
          </Field>
          <Field noMargin>
            <T.Caption lighterForeground>Title: The quick brown fox jumps over the lazy dog</T.Caption>
          </Field>
          <Field noMargin>
            <T.Header lighterForeground secondary>
              Header: The quick brown fox jumps over the lazy dog
            </T.Header>
          </Field>
          <Field noMargin>
            <T.Text lighterForeground secondary>
              Text: The quick brown fox jumps over the lazy dog
            </T.Text>
          </Field>
          <Field noMargin>
            <T.Caption lighterForeground secondary>
              Title: The quick brown fox jumps over the lazy dog
            </T.Caption>
          </Field>

          <Field noMargin>
            <T.Header lighterForeground highlight>
              Header: The quick brown fox jumps over the lazy dog
            </T.Header>
          </Field>
          <Field noMargin>
            <T.Text lighterForeground highlight>
              Text: The quick brown fox jumps over the lazy dog
            </T.Text>
          </Field>
          <Field noMargin>
            <T.Caption lighterForeground highlight>
              Title: The quick brown fox jumps over the lazy dog
            </T.Caption>
          </Field>
          <Field noMargin>
            <T.Header lighterForeground error>
              Header: The quick brown fox jumps over the lazy dog
            </T.Header>
          </Field>
          <Field noMargin>
            <T.Text lighterForeground error>
              Text: The quick brown fox jumps over the lazy dog
            </T.Text>
          </Field>
          <Field noMargin>
            <T.Caption lighterForeground error>
              Title: The quick brown fox jumps over the lazy dog
            </T.Caption>
          </Field>
        </div>
      </Block>
    );
  }
}

storiesOf('WidgetTypography', module).add('Typography', () => <Wrapper />);
