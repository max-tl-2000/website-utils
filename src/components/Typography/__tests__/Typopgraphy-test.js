/*
 * Copyright (c) 2022 Reva Technology Inc., all rights reserved.
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Licensed under the Elastic License 2.0; you may not use this file except
 * in compliance with the Elastic License 2.0.
 */

import React from 'react';

import { render, cleanup } from '@testing-library/react';
import * as T from '../Typopgraphy.js';

describe('Typography', () => {
  afterEach(cleanup);
  describe('Text', () => {
    it('should render a text element as a block', () => {
      const { container } = render(<T.Text>Some value to test</T.Text>);

      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render a text element as inline using a span', () => {
      const { container } = render(<T.Text inline>Some value to test</T.Text>);

      // verify tag element is an span inside the snaphsot
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
