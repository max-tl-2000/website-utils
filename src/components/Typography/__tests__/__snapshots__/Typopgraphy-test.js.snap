// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography Text should render a text element as a block 1`] = `
<p
  class="textElement Text primary noRaw"
  data-component="Text"
  data-text-element="true"
>
  Some value to test
</p>
`;

exports[`Typography Text should render a text element as inline using a span 1`] = `
<span
  class="textElement Text primary noRaw"
  data-component="Text"
  data-text-element="true"
>
  Some value to test
</span>
`;
